src/Classes/Utils/SpatialGrid.ts

```diff
@@ -0,0 +1,88 @@
+import { Particle } from "../Particle";
+import { ICoordinates } from "../../Interfaces/ICoordinates";
+import { IDimension } from "../../Interfaces/IDimension";
+
+
+// This class essentially works by interpreting all particles on the screen as a grid.
+// Grid cells are determined by dividing the width and height by the cell size. so 1920 / 10 = 19 cells of width
+// Particles are pushed into their respective cells with the same method.
+
+// For small particles this is great, but i expect there will need to be allowances for larger particles.
+// See https://i.stack.imgur.com/1Gx63.png.
+
+// Hence the addition of the radius option for selecting objects in adjacent cells. This also allows for broader detection for things like line drawing
+
+export class SpatialGrid {
+
+    cellSize: number;
+    widthSegment : number;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'widthSegment' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5342887)